### DISCUSSION TITLE
Search for type

### DISCUSSION THREAD

**Discussion by Machine-Maker** [mfussenegger/nvim-jdtls#494] [Q&A]
One of the big things I'm missing from IntelliJ is the ability to search for a class, not just in the project, but in all dependencies on the classpath. Is there a way I could do that, maybe a custom telescope picker or something to search for types on the classpath?

EDIT: Just a second thought, I also used the search for symbols quite a bit in intellij to find a method or field. Is that something that's possible as well?

**Comment by IlyasYOY** [comment]
@Machine-Maker Hello!

Yeah, you can do that. There is built-in support for Symbol Search in LSP. There is a function: `:h vim.lsp.buf.workspace_symbol()`. 

There is even more fluent solution: Telescope.

https://github.com/nvim-telescope/telescope.nvim#neovim-lsp-pickers

The one you are looking for is `builtin.lsp_dynamic_workspace_symbols`.
